hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -3900,4 +3900,25 @@ private void updateLayoutVersionInDB(OMLayoutVersionManager lvm,
   private BucketLayout getBucketLayout() {
     return BucketLayout.DEFAULT;
   }
+
+  /**
+   * Get supported Bucket Layouts based on upgrade state.
+   * Before finalization, the method returns an empty list. This leads to the
+   * new client's OzoneClientAdapter and CreateBucketHandler receiving an empty
+   * list - which causes them to continue with LEGACY layout.
+   * After finalization, the method returns a list with the new layouts.
+   * @return Supported Bucket Layouts.
+   */
+  public List<BucketLayout> getSupportedBucketLayouts() {
```

Comment:
@JyotinderSingh need to expose this function to `OzoneClientAdapter` and `CreateBucketHandler` so that client can send the default values.

Reply:
Thank you for the review, I have addressed this in my latest patch.